app/vlselect/traces/jaeger/model_test.go

```diff
@@ -0,0 +1,162 @@
+package jaeger
+
+import (
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage"
+	otelpb "github.com/VictoriaMetrics/VictoriaMetrics/lib/protoparser/opentelemetry/pb"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestFieldsToSpan(t *testing.T) {
+	f := func(input []logstorage.Field, want *span, wantErr error) {
+		t.Helper()
+		got, err := fieldsToSpan(input)
+		if !errors.Is(err, wantErr) && !strings.Contains(err.Error(), wantErr.Error()) {
+			t.Fatalf("fieldsToSpan() error = %v, want err: %v", err, wantErr)
+		}
+		cmpOpts := cmp.AllowUnexported(span{}, process{}, spanRef{}, keyValue{}, log{})
+		if !cmp.Equal(got, want, cmpOpts) {
+			t.Fatalf("fieldsToSpan() diff = %v", cmp.Diff(got, want, cmpOpts))
+		}
+	}
+
+	// case 1: empty
+	f([]logstorage.Field{}, nil, errors.New("invalid fields"))
+
+	// case 2: without span_id
+	fields := []logstorage.Field{
+		{Name: otelpb.TraceIDField, Value: "1234567890"},
+	}
+	f(fields, nil, errors.New("invalid fields"))
```

Comment:
This could be simplified even more: pass the error message as a string, in the success case, pass an empty string. That should be enough for the test.